lib/matplotlib/scale.py

```diff
@@ -103,14 +104,58 @@ def limit_range_for_scale(self, vmin, vmax, minpos):
         return vmin, vmax
 
 
+def handle_axis_parameter(init_func):
+    """
+    Decorator to handle the optional *axis* parameter in scale constructors.
+
+    This decorator ensures backward compatibility for scale classes that
+    previously required an *axis* parameter. It allows constructors to work
+    seamlessly with or without the *axis* parameter.
+
+    Parameters
+    ----------
+    init_func : callable
+        The original __init__ method of a scale class.
+
+    Returns
+    -------
+    callable
+        A wrapped version of *init_func* that handles the optional *axis*.
+
+    Notes
+    -----
+    If the wrapped constructor defines *axis* as its first argument, the
+    parameter is preserved when present. Otherwise, the value `None` is injected
+    as the first argument.
+
+    Examples
+    --------
+    >>> from matplotlib.scale import ScaleBase
+    >>> class CustomScale(ScaleBase):
+    ...     @handle_axis_parameter
+    ...     def __init__(self, axis, custom_param=1):
+    ...         self.custom_param = custom_param
+    """
+    @wraps(init_func)
+    def wrapper(self, *args, **kwargs):
+        if args and isinstance(args[0], mpl.axis.Axis):
+            return init_func(self, *args, **kwargs)
+        else:
+            # Remove 'axis' from kwargs to avoid double assignment
+            kwargs.pop('axis', None)
+            return init_func(self, None, *args, **kwargs)
+    return wrapper
+
+
 class LinearScale(ScaleBase):
     """
     The default linear scale.
     """
 
     name = 'linear'
 
-    def __init__(self, axis):
+    @handle_axis_parameter
+    def __init__(self, axis=None):
```

Comment:
Is there an advantage in making `axis` optional here? That should be handled be handled by the decorator. If there's no particular reason, then I'd prefer to keep the signature formally unchanged for now.